website/docs/components/code-block/partials/code/how-to-use.md

```diff
@@ -0,0 +1,126 @@
+## How to use this component
+
+The basic invocation requires a `@value` argument. The component encodes this argument before displaying it.
```

Comment:
I did a test with this string as `@value` and didn't recognize the `\n` escaped character as a newline: `package main\nimport fmt\nfunc main() {\n  fmt.Println(helloWorld)\n}`

Should we also explain how the newlines need to be added (and how are treated) by this argument?

(Something I would expect a developer will ask themselves, probably because there could be cases in which their code formatter will try to rework the "demo" code)

Reply:
> I did a test with this string as @value and didn't recognize the \n escaped character as a newline: package main\nimport fmt\nfunc main() {\n  fmt.Println(helloWorld)\n}

We generally don't convert `\n` into a new line in any of our components (such as Textarea) as it can have unexpected side effects if we do it at the component level. Instead, we rely on Ember's default behavior when dealing with strings. For instance, if you test with `<Hds::CodeBlock @value={{this.codeBlockContent}} />` and declare `@tracked codeBlockContent = "let codeLangs='JavaScript';\nconsole.log(codeLangs);";` in the controller/model it would resolve in new lines.

> Should we also explain how the newlines need to be added (and how are treated) by this argument?

Not sure. Wouldn't the examples be the reference for this? I'm happy to add more clarifications if needed, but not sure what the copy would be (_press enter for a new line_ may sound condescending).

Reply:
Probably I would do something like this:
```
## How to use this component

The basic invocation requires a `@value` argument. The component encodes this argument before displaying it.

_Notice: if the `\n` escape sequence is used in the `@value` string, it will **not** be automatically converted to a newline, as it can have unexpected side effects. _
```

Reply:
Thanks for the suggestion – added it [0a32f16](https://github.com/hashicorp/design-system/pull/1749/commits/0a32f164a849addc99bacd63412aa86e3d9c3d1c) in with the specification that this conversion doesn't happen if `/n` is used straight in the Handlebars syntax.

Reply:
Perfect!